src/hotspot/share/opto/vtransform.cpp

```diff
@@ -0,0 +1,441 @@
+}
+
+VTransformApplyResult VTransformElementWiseVectorNode::apply(const VLoopAnalyzer& vloop_analyzer, const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
+  Node* first = nodes().at(0);
+  uint  vlen = nodes().length();
+  int   opc  = first->Opcode();
+  BasicType bt = vloop_analyzer.types().velt_basic_type(first);
+
+  if (first->is_Cmp()) {
+    // Cmp + Bool -> VectorMaskCmp
+    // Handled by Bool / VTransformBoolVectorNode, so we do not generate any nodes here.
+    return VTransformApplyResult::make_empty();
+  }
+
+  assert(2 <= req() && req() <= 4, "Must have 1-3 inputs");
+  VectorNode* vn = nullptr;
+  Node* in1 =                find_transformed_input(1, vnode_idx_to_transformed_node);
+  Node* in2 = (req() >= 3) ? find_transformed_input(2, vnode_idx_to_transformed_node) : nullptr;
+  Node* in3 = (req() >= 4) ? find_transformed_input(3, vnode_idx_to_transformed_node) : nullptr;
+
+  if (first->is_CMove()) {
+    assert(req() == 4, "three inputs expected: mask, blend1, blend2");
+    vn = new VectorBlendNode(/* blend1 */ in2, /* blend2 */ in3, /* mask */ in1);
+  } else if (VectorNode::is_convert_opcode(opc)) {
+    assert(first->req() == 2 && req() == 2, "only one input expected");
+    int vopc = VectorCastNode::opcode(opc, in1->bottom_type()->is_vect()->element_basic_type());
+    vn = VectorCastNode::make(vopc, in1, bt, vlen);
+  } else if (VectorNode::can_use_RShiftI_instead_of_URShiftI(first, bt)) {
+    opc = Op_RShiftI;
+    vn = VectorNode::make(opc, in1, in2, vlen, bt);
+  } else if (VectorNode::is_scalar_op_that_returns_int_but_vector_op_returns_long(opc)) {
+    // The scalar operation was a long -> int operation.
+    // However, the vector operation is long -> long.
+    VectorNode* long_vn = VectorNode::make(opc, in1, nullptr, vlen, T_LONG);
+    register_new_node_from_vectorization(vloop_analyzer, long_vn, first);
+    // Cast long -> int, to mimic the scalar long -> int operation.
+    vn = VectorCastNode::make(Op_VectorCastL2X, long_vn, T_INT, vlen);
+  } else if (req() == 3 ||
+             VectorNode::is_scalar_unary_op_with_equal_input_and_output_types(opc)) {
+    assert(!VectorNode::is_roundopD(first) || in2->is_Con(), "rounding mode must be constant");
+    vn = VectorNode::make(opc, in1, in2, vlen, bt); // unary and binary
+  } else {
+    assert(req() == 4, "three inputs expected");
+    assert(opc == Op_FmaD ||
+           opc == Op_FmaF ||
+           opc == Op_SignumF ||
+           opc == Op_SignumD,
+           "element wise operation must be from this list");
+    vn = VectorNode::make(opc, in1, in2, in3, vlen, bt); // ternary
+  }
+
+  register_new_node_from_vectorization_and_replace_scalar_nodes(vloop_analyzer, vn);
+  return VTransformApplyResult::make_vector(vn, vlen, vn->length_in_bytes());
+}
+
+VTransformApplyResult VTransformBoolVectorNode::apply(const VLoopAnalyzer& vloop_analyzer, const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
```

Comment:
Maybe wrap this long line:
```suggestion
VTransformApplyResult VTransformBoolVectorNode::apply(const VLoopAnalyzer& vloop_analyzer, 
                                                      const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
```
Same at other apply methods.

Reply:
Good idea, applied it for all `apply` methods